### PR TITLE
added repository and service layers for book model

### PR DESCRIPTION
**Pull request by padu12**

### CODE REVIEW

File: src/main/java/com/kaziamyr/onlinebookstore/OnlineBookStoreApplication.java [hunk 1]
```diff
@@ -1,13 +1,37 @@
...
 import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.context.annotation.Bean;
 
 @SpringBootApplication
 public class OnlineBookStoreApplication {
+    private final BookService bookService;
+
+    public OnlineBookStoreApplication(BookService bookService) {
+        this.bookService = bookService;
+    }
```
Comment (Den-moroz) [inline src/main/java/com/kaziamyr/onlinebookstore/OnlineBookStoreApplication.java:19]: U should add annotation Autowired for it, but I'm not sure

File: .mvn/wrapper/maven-wrapper.properties [hunk 2]
```diff
@@ -0,0 +1,2 @@
+distributionUrl=https://repo.maven.apache.org/maven2/org/apache/maven/apache-maven/3.9.4/apache-maven-3.9.4-bin.zip
+wrapperUrl=https://repo.maven.apache.org/maven2/org/apache/maven/wrapper/maven-wrapper/3.2.0/maven-wrapper-3.2.0.jar
```
Comment (Den-moroz) [inline .mvn/wrapper/maven-wrapper.properties:2]: I think you shouldn't push .mvn folder in pull-request

**Review by Den-moroz — CHANGES_REQUESTED** [review]
Good solution but need to fix some moment

File: src/main/java/com/kaziamyr/onlinebookstore/OnlineBookStoreApplication.java (hunk 1 above)
Comment (vovasalyha) [reply src/main/java/com/kaziamyr/onlinebookstore/OnlineBookStoreApplication.java:19]: It's no a requirement to mark a constructor with `Autowired`. Spring will understand it anyway. And if I'm not mistaken, you can pass bean directly to commandLineRunner method params.

File: src/main/java/com/kaziamyr/onlinebookstore/repository/impl/BookRepositoryImpl.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+        try {
+            session = sessionFactory.openSession();
+            transaction = session.beginTransaction();
+            session.persist(book);
+            transaction.commit();
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new RuntimeException("Can't add book: " + book);
```
Comment (vovasalyha) [inline src/main/java/com/kaziamyr/onlinebookstore/repository/impl/BookRepositoryImpl.java]: Throwing a general RuntimeException is not best practice.

File: src/main/java/com/kaziamyr/onlinebookstore/repository/impl/BookRepositoryImpl.java [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+        this.sessionFactory = sessionFactory;
+    }
+
+    @Override
+    public Book save(Book book) {
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = sessionFactory.openSession();
+            transaction = session.beginTransaction();
```
Comment (vovasalyha) [inline src/main/java/com/kaziamyr/onlinebookstore/repository/impl/BookRepositoryImpl.java:27]: You have Spring data dependency, I think you could use `@Transactional` and simply use `Session` with try-with-resources without manual transaction handling.

**Review by vovasalyha — CHANGES_REQUESTED** [review]
Great work! Please take a look on my comments too.

**Review by vovasalyha — APPROVED** [review]

**Pull request merged**